### PR TITLE
sort 10 sample / all unique values list in the expression builder widget

### PR DESCRIPTION
**Pull request by nirvn**
## Description
Before vs. proposed:
screenshot from 2018-01-12 12-10-57

It'd be a shame not to 😉 

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by lbartoletti** [comment]
Good idea!

I'm wondering if we can have some improvements:
- An option to sort by Increase, Decrease.
- Sort by count of use of the values.

**Comment by nirvn** [comment]
@lbartoletti , those ideas can be visited in 3.2. 

This PR focuses on harmonizing the Filter Query Builder and the Expression Buildget Widget amid our hard freeze period.

File: src/gui/qgsexpressionbuilderwidget.cpp [hunk 1]
```diff
@@ -349,7 +349,8 @@ void QgsExpressionBuilderWidget::fillFieldValues( const QString &fieldName, int
     return;
 
   QStringList strValues;
-  QSet<QVariant> values = mLayer->uniqueValues( fieldIndex, countLimit );
+  QList<QVariant> values = mLayer->uniqueValues( fieldIndex, countLimit ).toList();
+  qSort( values );
```
Comment (nyalldawson) [inline src/gui/qgsexpressionbuilderwidget.cpp]: Can you use the std:: sort methods instead? qSort is dead.

**Comment by nirvn** [comment]
@nyalldawson , good catch, done.

**Pull request merged**